backend/src/LibExecution/TypeChecker.fs

```diff
@@ -179,6 +184,18 @@ let raiseValueNotExpectedType
   : 'a =
   raiseRTE source (ValueNotExpectedType(dv, typ, context) |> Error.toRuntimeError)
 
+let raiseFnValResultNotExpectedType
+  (source : DvalSource)
+  (dv : Dval)
+  (typ : TypeReference)
```

Comment:
small nitpick, maybe swap dv (actual) and typ (expected) here?

Reply:
Actual comes before expected, no?

Reply:
Seems to differ in different worlds. I'm just really used to the .NET world where it's generally expected first. That's also what I've seen in F# libraries. https://chat.openai.com/share/91365a6c-3f59-4d2a-8b51-b710f1921415 for reference